src/python/pants_release/generate_github_workflows.py

```diff
@@ -1055,17 +1055,46 @@ def release_jobs_and_inputs() -> tuple[Jobs, dict[str, Any]]:
                         """
                     ),
                 },
+                {
+                    "name": "Make GitHub Release",
+                    "if": f"{IS_PANTS_OWNER} && steps.get_info.outputs.is-release == 'true'",
+                    "run": dedent(
+                        """\
+                        RELEASE_TAG=${{ steps.get_info.outputs.build-ref }}
+                        RELEASE_VERSION="${RELEASE_TAG#release_}"
+
+                        # NB: This could be a re-run of a release, in the event a job/step failed.
+                        if gh release view $RELEASE_TAG ; then
+                            exit 0
+                        fi
+
+                        GH_RELEASE_ARGS=("--notes" "")
+                        GH_RELEASE_ARGS+=("--title" "$RELEASE_TAG")
+                        if [[ $RELEASE_VERSION =~ [[:alpha:]] ]]; then
+                            GH_RELEASE_ARGS+=("--prerelease")
+                        else
+                            STABLE_RELEASE_TAGS=$(gh api -X GET -F per_page=100 /repos/{owner}/{repo}/releases --jq '.[].tag_name | sub("^release_"; "")')
+                            LATEST_TAG=$(echo "$STABLE_RELEASE_TAGS $RELEASE_TAG" | tr ' ' '\\n' | sort --version-sort | tail -n 1)
+                            if [[ $RELEASE_TAG == $LATEST_TAG ]]; then
+                                GH_RELEASE_ARGS+=("--latest")
+                            fi
+                        fi
+
+                        gh release create "$RELEASE_TAG" "${GH_RELEASE_ARGS[@]}"
```

Comment:
There's a chance of a race condition here: the release is created as published release, but has nothing in it, so if someone notices the release and updates their pants.toml, they'll get weird results.

This would be 'particularly' bad if the release process fails, so the partial-release is sitting around for hours/days while people get release going again.

One option to handle this edge case would be creating the release in `--draft`, and then after everything has succeeded, run `gh release edit ... --draft=false` to publish. Thoughts?

Reply:
Hmmmmmmmm. Yeah I like it.